### PR TITLE
CORE-10333: Remove Kotlin "utility" functions from application module.

### PR DESCRIPTION
**Pull request by chrisr3**
Remove the remaining Kotlin function from the `:application` API module.

Also provide an API to parse JSON data into a `Map`, because the existing API for doing this was not very "Java-friendly".

### CODE REVIEW

File: application/src/main/kotlin/net/corda/v5/application/messaging/FlowSessionUtils.kt [hunk 1]
```diff
@@ -1,37 +0,0 @@
...
- * Note that this function is not just a simple send+receive pair: it is more efficient and more correct to use this
- * when you expect to do a message swap than do use [FlowSession.send] and then [FlowSession.receive] in turn.
- *
- * @param R The data type received from the counterparty.
- * @param payload The data sent to the counterparty.
- *
- * @return The received data [R]
- */
-@Suspendable
-inline fun <reified R : Any> FlowSession.sendAndReceive(payload: Any): R {
```
Comment (mnesbit) [inline application/src/main/kotlin/net/corda/v5/application/messaging/FlowSessionUtils.kt:19]: Ouch that is going to hurt!

Comment (chrisr3) [reply application/src/main/kotlin/net/corda/v5/application/messaging/FlowSessionUtils.kt:19]: Yes. Kotlin developer will now feel Java developers' pain.

File: application/src/main/java/net/corda/v5/application/flows/ClientRequestBody.java [hunk 2]
```diff
@@ -47,4 +48,20 @@ public interface ClientRequestBody {
...
+     * The selected {@link MarshallingService} will determine what format data is returned.
+     *
+     * @param marshallingService The {@link MarshallingService} to use to deserialize this request body.
+     * @param keyClass The class to deserialize the key data into.
+     * @param valueClass The class to deserialize the value data into.
+     *
+     * @return A map of instances of the value class populated by the provided input data.
+     */
+    @NotNull
+    <K, V> Map<K, V> getRequestBodyAsMap(@NotNull MarshallingService marshallingService,
```
Comment (lankydan) [inline application/src/main/java/net/corda/v5/application/flows/ClientRequestBody.java]: Can we reformat this method.

Comment (chrisr3) [reply application/src/main/java/net/corda/v5/application/flows/ClientRequestBody.java]: Reformat "how"?

Comment (chrisr3) [reply application/src/main/java/net/corda/v5/application/flows/ClientRequestBody.java]: :+1:

**Review by mcgovc — APPROVED** [review]
LGTM - build file changes only

**Review by lankydan — APPROVED** [review]

**Pull request merged**